### PR TITLE
fix: make sure scale color works for sparkline

### PR DESCRIPTION
**Pull request by JennChao**
### Updates
User should be able to set scale color for their charts, including sparkline. Previously scale color doesn't work on sparkline. This PR if created for fixing that issue.

### Demo screenshot or recording
圖片

### Review checklist (for reviewers only)
- [ ] Demos all features
- [ ] Documented/annotated
- [ ] Matches UI/UX specs
- [ ] Meets the [code style guide](https://github.com/carbon-design-system/carbon-charts/wiki/Code-style-guide)
- [ ] Accessible
- [ ] Mobile first (responsive)
- [ ] RTL support (bidirectional text)
- [ ] Performant (limited bloat)

### CODE REVIEW

File: packages/core/src/components/graphs/area.ts [hunk 1]
```diff
@@ -101,11 +101,15 @@ export class Area extends Component {
...
+			this.model.getOptions(),
+			"color",
+			"scale");
 		const colorValue = strokePathElement
 			? getComputedStyle(strokePathElement, null).getPropertyValue(
 					"stroke"
-			  )
-			: null;
+			)
+			: isGradientAllowed ? sparklineColorObject[Object.keys(sparklineColorObject)[0]] : null;
```
Comment (theiliad) [inline packages/core/src/components/graphs/area.ts:112]: very confusing code here. Are we able to clean this up? Also, shouldn't `isGradientAllowed` be check before checking the stroke?

**Review by theiliad — CHANGES_REQUESTED** [review]

File: packages/core/src/components/graphs/area.ts (hunk 1 above)
Comment (JennChao) [reply packages/core/src/components/graphs/area.ts]: `isGradientAllowed` is now being checked before checking stroke.

**Comment by sophiiae** [comment]
Screen Shot 2020-11-18 at 2 28 52 PM
sparkline doesn't look right for the default color palette in this PR. 
The gradient is missing.

**Comment by JennChao** [comment]
> Screen Shot 2020-11-18 at 2 28 52 PM
> sparkline doesn't look right for the default color palette in this PR.
> The gradient is missing.

@sophiiae 
The default color palette is now working on sparkline. Please help to review this PR again. Thanks!

**Comment by theiliad** [comment]
> > Screen Shot 2020-11-18 at 2 28 52 PM
> > sparkline doesn't look right for the default color palette in this PR.
> > The gradient is missing.
> 
> @sophiiae
> The default color palette is now working on sparkline. Please help to review this PR again. Thanks!

Hi, the gradient seems to be all grey
image

**Comment by JennChao** [comment]
> > > Screen Shot 2020-11-18 at 2 28 52 PM
> > > sparkline doesn't look right for the default color palette in this PR.
> > > The gradient is missing.
> > 
> > 
> > @sophiiae
> > The default color palette is now working on sparkline. Please help to review this PR again. Thanks!
> 
> Hi, the gradient seems to be all grey
> image

@theiliad 
The reason why we got a gray gradient here is because of the following settings.
圖片
This PR is actually created for enabling scale color applied to sparkline. Previously it doesn't work...

Should I remove the scale color setting in demo storybook so you can see default color palette?

**Comment by sophiiae** [comment]
> > > > Screen Shot 2020-11-18 at 2 28 52 PM
> > > > sparkline doesn't look right for the default color palette in this PR.
> > > > The gradient is missing.
> > > 
> > > 
> > > @sophiiae
> > > The default color palette is now working on sparkline. Please help to review this PR again. Thanks!
> > 
> > 
> > Hi, the gradient seems to be all grey
> > image
> 
> @theiliad
> The reason why we got a gray gradient here is because of the following settings.
> 圖片
> This PR is actually created for enabling scale color applied to sparkline. Previously it doesn't work...
> 
> Should I remove the scale color setting in demo storybook so you can see default color palette?

The color palette is working now, but it may not be a good idea to set grey as demo color.

**Comment by sophiiae** [comment]
@JennChao this PR has conflicts.

**Comment by theiliad** [comment]
> > > > > Screen Shot 2020-11-18 at 2 28 52 PM
> > > > > sparkline doesn't look right for the default color palette in this PR.
> > > > > The gradient is missing.
> > > > 
> > > > 
> > > > @sophiiae
> > > > The default color palette is now working on sparkline. Please help to review this PR again. Thanks!
> > > 
> > > 
> > > Hi, the gradient seems to be all grey
> > > image
> > 
> > 
> > @theiliad
> > The reason why we got a gray gradient here is because of the following settings.
> > 圖片
> > This PR is actually created for enabling scale color applied to sparkline. Previously it doesn't work...
> > Should I remove the scale color setting in demo storybook so you can see default color palette?
> 
> The color palette is working now, but it may not be a good idea to set grey as demo color.

yes we should probably leave our demo working with the color palette rather than setting the color to grey.

**Comment by JennChao** [comment]
> @JennChao this PR has conflicts.

Conflicts have already been resolved.

**Comment by JennChao** [comment]
> > > > > > Screen Shot 2020-11-18 at 2 28 52 PM
> > > > > > sparkline doesn't look right for the default color palette in this PR.
> > > > > > The gradient is missing.
> > > > > 
> > > > > 
> > > > > @sophiiae
> > > > > The default color palette is now working on sparkline. Please help to review this PR again. Thanks!
> > > > 
> > > > 
> > > > Hi, the gradient seems to be all grey
> > > > image
> > > 
> > > 
> > > @theiliad
> > > The reason why we got a gray gradient here is because of the following settings.
> > > 圖片
> > > This PR is actually created for enabling scale color applied to sparkline. Previously it doesn't work...
> > > Should I remove the scale color setting in demo storybook so you can see default color palette?
> > 
> > 
> > The color palette is working now, but it may not be a good idea to set grey as demo color.
> 
> yes we should probably leave our demo working with the color palette rather than setting the color to grey.

Sparkline now use default color palette.

**Comment by JennChao** [comment]
@sophiiae @theiliad 
Please help to review this PR again. Thanks!

**Review by sophiiae — DISMISSED** [review]

**Review by cal-smith — DISMISSED** [review]

**Review by theiliad — APPROVED** [review]

**Review by cal-smith — APPROVED** [review]

**Pull request merged**